Packs/Gmail/Integrations/Gmail/Gmail.py

```diff
@@ -1146,55 +1256,69 @@ def get_user_tokens(user_id):
     return result.get('items', [])
 
 
-def search_all_mailboxes(receive_only_accounts):
-    users_next_page_token = None
-    service = get_service('admin', 'directory_v1')
-    while True:
-        command_args = {
-            'maxResults': 100,
-            'domain': ADMIN_EMAIL.split('@')[1],  # type: ignore
-            'pageToken': users_next_page_token
-        }
+def search_in_mailboxes(list_accounts: list[str], receive_only_accounts: bool) -> None:
 
-        result = service.users().list(**command_args).execute()
-        users_count = len(result['users'])
-        users_next_page_token = result.get('nextPageToken')
+    futures: list = []
+    entries: list = []
+    with concurrent.futures.ThreadPoolExecutor() as executor:
+        for user in list_accounts:
+            futures.append(executor.submit(search_command, mailbox=user,
+                                           receive_only_accounts=receive_only_accounts))
+        for account in concurrent.futures.as_completed(futures):
+            if found := account.result():
```

Comment:
what if it's not found?

Reply:
It returns None so I don't collect the results in that case

Reply:
please mention it in the docstring of this method